react-forms/src/InputWithButton.tsx

```diff
@@ -0,0 +1,142 @@
+// ******************************************************************************************************
+//  Input.tsx - Gbtc
+//
+//  Copyright � 2024, Grid Protection Alliance.  All Rights Reserved.
+//
+//  Licensed to the Grid Protection Alliance (GPA) under one or more contributor license agreements. See
+//  the NOTICE file distributed with this work for additional information regarding copyright ownership.
+//  The GPA licenses this file to you under the MIT License (MIT), the "License"; you may not use this
+//  file except in compliance with the License. You may obtain a copy of the License at:
+//
+//      http://opensource.org/licenses/MIT
+//
+//  Unless agreed to in writing, the subject software distributed under the License is distributed on an
+//  "AS-IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. Refer to the
+//  License for the specific language governing permissions and limitations.
+//
+//  Code Modification History:
+//  ----------------------------------------------------------------------------------------------------
+//  0502/2024 - Preston Crawford
+//       Generated original version of source code.
+//
+// ******************************************************************************************************
+
+import * as React from 'react';
+import HelperMessage from './HelperMessage';
+import { CreateGuid, IsInteger, IsNumber } from '@gpa-gemstone/helper-functions'
+
+interface IProps<T> {
+    Record: T;
+    Field: keyof T;
+    Setter: (record: T) => void;
+    Valid: (field: keyof T) => boolean;
+    InputLabel?: string;
```

Comment:
```suggestion
    Label?: string;
```

Nitpicky but naming of props does not need to include Input if it is for the whole component